docs/src/main/asciidoc/spring-data-support.adoc

```diff
@@ -91,7 +113,162 @@ spring:
+
+    Flux<User> findByFirstName(String firstName);
+}
+----
+
+So far ReactiveCosmosRepository provides basic save, delete and find operations. More operations will be supported later.
+
+==== Create an Application class
+Here create an application class with all the components
+
+----
+@SpringBootApplication
+public class CosmosSampleApplication implements CommandLineRunner {
+
+   private static final Logger LOGGER = LoggerFactory.getLogger(CosmosSampleApplication.class);
+
+	@Autowired
+	private UserRepository repository;
+
+	@Autowired
+	private CosmosProperties properties;
+
+	public static void main(String[] args) {
+		SpringApplication.run(CosmosSampleApplication.class, args);
+	}
+
+	public void run(String... var1) {
+		final User testUser = new User("testId", "testFirstName",
+				"testLastName", "test address line one");
+
+		// Save the User class to Azure Cosmos DB database.
+		final Mono<User> saveUserMono = repository.save(testUser);
+
+		final Flux<User> firstNameUserFlux = repository.findByFirstName("testFirstName");
+
+		//  Nothing happens until we subscribe to these Monos.
+		//  findById will not return the user as user is not present.
+		final Mono<User> findByIdMono = repository.findById(testUser.getId());
+		final User findByIdUser = findByIdMono.block();
+		Assert.isNull(findByIdUser, "User must be null");
+
+		final User savedUser = saveUserMono.block();
+		Assert.state(savedUser != null, "Saved user must not be null");
+		Assert.state(savedUser.getFirstName().equals(testUser.getFirstName()),
+				"Saved user first name doesn't match");
+
+		firstNameUserFlux.collectList().block();
+
+		final Optional<User> optionalUserResult = repository.findById(testUser.getId()).blockOptional();
+		Assert.isTrue(optionalUserResult.isPresent(), "Cannot find user.");
+
+		final User result = optionalUserResult.get();
+		Assert.state(result.getFirstName().equals(testUser.getFirstName()),
+				"query result firstName doesn't match!");
+		Assert.state(result.getLastName().equals(testUser.getLastName()),
+				"query result lastName doesn't match!");
+		LOGGER.info("findOne in User collection get result: {}", result.toString());
+
+	}
+
+	@PostConstruct
+	public void setup() {
+		// For this example, remove all of the existing records.
+		this.repository.deleteAll().block();
+	}
+}
+----
+
+Autowired UserRepository interface, then can do save, delete and find operations.
+
+
+
+=== Next steps
+Besides using this Azure Cosmos DB Spring Boot Starter, you can directly use Spring Data for Azure Cosmos DB package for more complex scenarios. Please refer to link:https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/cosmos/azure-spring-data-cosmos[Spring Data for Azure Cosmos DB] for more details.
+
+=== More Sample
+Please refer to link:https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_4.0-beta.2/cosmos[azure-spring-boot-samples] for more details.
```

Comment:
This link is also broken. https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_4.0-beta.2/cosmos